alica_engine/src/engine/Logger.cpp

```diff
@@ -17,23 +17,21 @@
 
 #include <essentials/FileSystem.h>
 
-//#define ALICA_DEBUG_LEVEL_ALL
-#include <alica_common_config/debug_output.h>
-
 using std::endl;
 
 namespace alica
 {
 using std::to_string;
 
-Logger::Logger(AlicaEngine* ae)
+Logger::Logger(AlicaEngine* ae, IAlicaLogger& logger)
         : _ae(ae)
         , _fileWriter()
         , _itCount(0)
         , _active(false)
         , _receivedEvent(false)
         , _inIteration(false)
         , _logging(false)
+        , _logger(logger)
```

Comment:
should rename either this variable or this class. Having a logger member variable in a Logger class is weird